plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -429,11 +429,21 @@ private TestExecutionResult testPluginAgainst(MavenCoordinates coreCoordinates,
         return newUpdateSiteData(us, JSONObject.fromObject(json));
 	}
 
-    private UpdateSite.Data scanWAR(File war) throws IOException {
+    /**
+     * Scans through a WAR file, accumulating plugin information
+     * @param war WAR to scan
+     * @param pluginGroupIds Map pluginName to groupId if set in the manifest, MUTATED IN THE EXECUTION
+     * @return Update center data
+     * @throws IOException
+     */
+    private UpdateSite.Data scanWAR(File war, Map<String, String> pluginGroupIds) throws IOException {
         JSONObject top = new JSONObject();
         top.put("id", DEFAULT_SOURCE_ID);
         JSONObject plugins = new JSONObject();
         JarFile jf = new JarFile(war);
+        if (pluginGroupIds == null) {
+            pluginGroupIds = new HashMap<String, String>();
```

Comment:
Why is this parameter not just `@Nonnull`? It was already initialized by the caller.